### PR TITLE
sort out the sensor one readings for the internal use case of GET Events

### PR DESCRIPTION
**Pull request by Baalmart**
# sort out the sensor one readings for the internal use case of GET Events

**_WHAT DOES THIS PR DO?_**

- [x] Sort out the sensor one readings

Showcase the sensor one readings for the internal use case. Differentiates the sensor ones readings (`s1_pm2_5` and `s1_pm10`) from the average readings of the sensor one readings (`pm10` and `pm2.5`) in each devices.
If does this change through the introduction of new fields that correspond to the sensor one readings. And leaves the `pm2_5` and `pm10` to be for the average calibrated readings that are usually consumed by the external consumers.

- s1_pm2_5
- s1_pm10

- [x] Optimise performance by reducing the payload for the metadata (`siteDetails` and `deviceDetails`)

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/hotfix-create-event

**_HOW DO I TEST OUT THIS PR?_**
README, staging

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [GET events](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/device-registry/events#get-events)

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A

**_SCREENSHOTS?_**
Screenshot 2021-11-21 at 23 24 47

Screenshot 2021-11-22 at 03 20 21

Screenshot 2021-11-22 at 03 23 39

### CODE REVIEW

**Review by NoahNsimbe — COMMENTED** [review]
@Baalmart please add the following to the siteDetails 
- region
- country
- district

**Comment by Baalmart** [comment]
> @Baalmart please add the following to the siteDetails
> 
> * region
> * country
> * district

This has been addressed. Thanks @NoahNsimbe

File: src/device-registry/models/Event.js [hunk 1]
```diff
@@ -355,45 +357,105 @@ eventSchema.statics = {
...
+      if (metadata === "device" || !metadata) {
+        foreignField = "name";
+      }
+
+      from = "devices";
+      _as = "_deviceDetails";
+      as = "deviceDetails";
+      elementAtIndex0 = { $arrayElemAt: ["$deviceDetails", 0] };
+
+      projection[as] = {};
```
Comment (Baalmart) [inline src/device-registry/models/Event.js:375]: Please note that there is code duplication in this PR, there is no good application of "code reuse". Delighted to inform you that this issue is being handled by another ongoing [refactoring (pagination) PR](https://github.com/airqo-platform/AirQo-api/pull/674). @ja-odur @NoahNsimbe

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Review by ja-odur — APPROVED** [review]
LGTM

**Pull request merged**